hackoregon/team-budget
# First automated test [depends on PR #62]

This is the first automated test to be added to this project.

It verifies that the web server hosting the endpoint is up and responding.  It does not determine if a valid response is sent by the API (this can be a separate test or many).

This test will be helpful for Travis automated builds to verify that nothing has taken the API completely offline.  (Necessary but not sufficient testing.)

I copied shamelessly from the tests being developed by the Housing backend team:
https://github.com/hackoregon/housing-backend/blob/master/backend/housing_backend/tests/test_API.py

# How to Test
- `git pull`
- `git checkout add-one-test`
- `./budget_proj/bin/test-proj.sh`

You should observe the following output:

```
$ ./budget_proj/bin/test-proj.sh 
Running test_proj.sh...
Creating test database for alias 'default'...
.
----------------------------------------------------------------------
Ran 1 test in 0.164s

OK
Destroying test database for alias 'default'...
```


# Note
I developed this on top of PR #62.  It could be decoupled but the additional work wasn't worth it to me, and this PR should result in a trivial change one PR #62 is merged.
